### ISSUE TITLE
Stdlib function to transform sketches/solids

### ISSUE THREAD

**Issue by adamchalmers** [KittyCAD/modeling-app#4373]
# Background

Say you're trying to draw a square in KCL. Pretty straightforward.

```
fn square = (length) => {
  l = length / 2
  p0 = [-l, -l]
  p1 = [-l, l]
  p2 = [l, l]
  p3 = [l, -l]
  return startSketchAt(p0)
  |> lineTo(p1, %)
  |> lineTo(p2, %)
  |> lineTo(p3, %)
  |> close(%)
}
square(20)
```

This draws a square with side length 20. The square's center is at [0, 0]. 

# Problem

What if you want to make the center a parameter? The code becomes more complex.

```
fn squareAt = (length, center) => {
  l = length / 2
  x = center[0]
  y = center[1]
  p0 = [-l + x, -l + y]
  p1 = [-l + x, l + y]
  p2 = [l + x, l + y]
  p3 = [l + x, -l + y]
  return startSketchAt(p0)
  |> lineTo(p1, %)
  |> lineTo(p2, %)
  |> lineTo(p3, %)
  |> close(%)
}
```

We have to do a lot of manual math.

# Solution

This could be dramatically simplified if we wrote a `translate(point, sketch)` function that took a sketch (or solid?) and translated it in some direction. 

```
fn squareAt = (length, center) => {
  square(length) |> translate(center)
}
```

But that might be a bit too specific. I think @benjamaan476 has talked about reusing the Transformation we use in the "pattern transform" API for single objects. I think we should do that!

```
fn squareAt = (length, center) => {
  square(length) |> transform({rotate: 45, translate: [10, 0, 0]}, %)
}
```

This would DRAMATICALLY simplify my helical gear.

![Image](https://github.com/user-attachments/assets/749e4755-afd3-4caa-9065-f6fe86e349a1)

I know the code diff doesn't look very dramatic, but getting that offset angle correct took a lot of very careful math that Ben and I had to collaborate on, it probably would have saved me an hour of work.

**Comment by benjamaan476** [comment]
Yeah that is definitely doable! My transform work that I sketched out was for Objects so there is no reason it can't be used for Solid2ds. That's basically what we do in patternTransforms. Shouldn't be too difficult to expose

**Comment by benjamaan476** [comment]
It occurred to me over the weekend that you can probably gain a lot of flexibility by adding a skew option to this. You don't do this in normal rendering as it is considered a distortion, which you don't want. But thinking of @jgomez720 's problems in his kcl doc with trying to make that equal sided diamond for the base plate, you could skew an equal length square.

**Comment by jtran** [comment]
`translate()`, `rotate()`, and `scale()` were added in #5067. They work on solids only.

**Comment by benjamaan476** [comment]
Why solid only? The endpoint works on paths, planes, solid2d and solid3d

**Comment by jgomez720** [comment]
I like that we have the option for all of those, but we don't move sketches, planes, and paths. To be honest, I don't think we even have the option to in our traditional CAD. If we need to move those, we go into the source and edit the parameters.

**Comment by franknoirot** [comment]
Would we call this closed as complete?

**Comment by benjamaan476** [comment]
I feel like @adamchalmers has given a compelling example for editing a sketch, just making everyone aware it is something that we could add

**Comment by benjamaan476** [comment]
Thinking about this more it feels very useful to be able to transform sketches, e.g. if you wanted to create a twisted loft is would be as simple as cloning the base sketch, translating it up and then rotating it to get the twist, no need to calculate where all the new vertices will be by hand and no room for error by missing one etc.

**Comment by jgomez720** [comment]
@benjamaan476 I didn't think of that, that's a fantastic idea. I'm much more in favor of transforming of 2D objects now. Now that I have had more time to think, I have also come up with good use cases for transforming of sketches and paths

**Comment by jessfraz** [comment]
okay ill allow it on sketches and solids only
